### PR TITLE
Using Singles, Fee Asset Id and ServiceResponse

### PR DESCRIPTION
**Pull request by amirchohan**

### CODE REVIEW

File: src/main/java/com/gsr/bitrue/order/poller/TradePoller.java [hunk 1]
```diff
@@ -26,37 +35,67 @@
...
 
 
-  public Option<List<OrderExecution>> poll() {
+  public Try<List<OrderExecution>> poll() {
+    return Try.of(() -> {
+
+      List<OrderExecution> executions = new ArrayList<>();
+      // get last trade ts from ogw
+      val lastTradeTs = driver.getLastTradeTs().map(Long::valueOf).getOrElse(0L);
+      long sinceTime = Try.of(() -> lastTradeTs/1000000).getOrElse(0L);
```
Comment (flopezlasanta) [inline src/main/java/com/gsr/bitrue/order/poller/TradePoller.java]: TimeUnit to convert instead of this division?

Comment (flopezlasanta) [reply src/main/java/com/gsr/bitrue/order/poller/TradePoller.java]: also, why wrapping within a Try?

File: src/main/java/com/gsr/bitrue/order/poller/TradePoller.java [hunk 2]
```diff
@@ -26,37 +35,67 @@
...
-      log.debug("fetching trades (since:{})", sinceTime);
+        if (response.isSuccess()) {
+          val trades = response.getPayload().stream().map(this::toExecution).collect(Collectors.toList());
+          executions.addAll(trades);
 
-      val tradesResponse = service.getTrades(sinceTime, TRADES_LIMIT).blockingSingle(new ArrayList<>());
+          if (trades.size() < TRADES_LIMIT) {
+            break;
+          }
+          sinceTime = response.getPayload().get(trades.size() - 1).getTime();
```
Comment (flopezlasanta) [inline src/main/java/com/gsr/bitrue/order/poller/TradePoller.java]: you can query the time from trades (the last one) directly

File: src/main/java/com/gsr/bitrue/order/poller/TradePoller.java [hunk 3]
```diff
@@ -26,37 +35,67 @@
...
+        .tradeId(exchangeTrade.getId().toString())
+        .price(exchangeTrade.getPrice())
+        .quantity(exchangeTrade.getQty())
+        .side(exchangeTrade.isBuyer() ? Side.BUY : Side.SELL)
+        .status(ExecutionStatus.CONFIRMED)
+        .tradeTime(tradeTime)
+        .receivedTime(TimeUtils.nowUTC())
+        .type(exchangeTrade.isMaker() ? ExecutionType.MAKER : ExecutionType.TAKER)
+        .fee(exchangeTrade.getCommission())
+        .feeAssetId(feeAssetId)
```
Comment (flopezlasanta) [inline src/main/java/com/gsr/bitrue/order/poller/TradePoller.java]: great!

File: src/main/java/com/gsr/bitrue/order/poller/TradePoller.java (hunk 1 above)
Comment (amirchohan) [reply src/main/java/com/gsr/bitrue/order/poller/TradePoller.java]: done. It was wrapped within Try to catch division of zero

File: src/main/java/com/gsr/bitrue/order/poller/TradePoller.java (hunk 2 above)
Comment (amirchohan) [reply src/main/java/com/gsr/bitrue/order/poller/TradePoller.java]: that'll require converting time from nano to milli :)

**Pull request merged**